packages/playwright-crawler/src/internals/adaptive-playwright-crawler.ts

```diff
@@ -0,0 +1,263 @@
+import { addTimeoutToPromise } from '@apify/timeout';
+import { BasicCrawler } from '@crawlee/basic';
+import type { Source, Configuration, RecordOptions, RestrictedCrawlingContext } from '@crawlee/core';
+import { KeyValueStore } from '@crawlee/core';
+import type { Awaitable, Dictionary } from '@crawlee/types';
+import { extractUrlsFromCheerio } from '@crawlee/utils';
+import type { Cheerio, Document } from 'cheerio';
+import { load } from 'cheerio';
+import { cheerioPortadom, playwrightLocatorPortadom, type CheerioPortadom, type PlaywrightLocatorPortadom } from 'portadom';
+
+import type { PlaywrightCrawlerOptions, PlaywrightCrawlingContext } from './playwright-crawler';
+import { PlaywrightCrawler } from './playwright-crawler';
+import { RenderingTypePredictor, type RenderingType } from './rendering-type-prediction';
+
+type Result<TResult> = NonNullable<{result: TResult; ok: true} | {error: unknown; ok: false}>
+
+class RequestHandlerResult {
```

Comment:
This should probably go into a shared module such as `crawler_commons`. Any better ideas?

Reply:
@B4nan says anything under core